python/lsst/ts/cRIOpy/GUI/ApplicationControlWidget.py

```diff
@@ -38,6 +39,59 @@
 from lsst.ts.idl.enums.MTM1M3 import DetailedState
 
 
+class HPWarnings:
+    def __init__(self, m1m3):
+        self.faultHigh = self.faultLow = None
+        self.warningHigh = self.warningLow = None
+
+        self._faultLow = self._faultLowRaising = None
+
+        m1m3.hardpointActuatorSettings.connect(self.hardpointActuatorSettings)
+
+    def setState(self, state):
+        """Change low limits according to sensed state.
+
+        Note
+        ----
+        As that needs to be called before toottips are set, do not hook that
+        directly to detailedState signal.
+
+        Parameters
+        ----------
+        state : `int`
```

Comment:
enum or int?

Reply:
It's int from SALObj

Reply:
It isn't. Changed to enum.intEnum, Thanks